index.bs

```diff
@@ -91,66 +113,67 @@ dictionary CSSStyleSheetInit {
     <dd>
     	1. Let |sheet| be the stylesheet on which this function is called on.
 		2. If |sheet|'s [=constructed flag=] and [=disallow modification flag=] is set, throw "{{NotAllowedError}}" {{DOMException}}.
-		3. If |rule| is an <a spec=css-cascade-4>@import</a> rule and |sheet|'s [=constructed flag=] is set, throw "{{NotAllowedError}}" {{DOMException}}.
+        3. <a>Parse a rule</a> from |rule|. If the result is an <a>@import</a> rule and |sheet|'s [=constructed flag=] is set, throw "{{NotAllowedError}}" {{DOMException}}.
 		4. (The rest of the algorithm remains as in CSSOM)
 	</dd>
 
-	<dt><dfn method for=CSSStyleSheet lt="deleteRule(text)">deleteRule(text)</dfn></dt>
+    <dt><dfn method for=CSSStyleSheet>deleteRule(|index|)</dfn></dt>
     <dd>
     	1. Let |sheet| be the stylesheet on which this function is called on.
 		2. If |sheet|'s [=constructed flag=] and [=disallow modification flag=] is set, throw "{{NotAllowedError}}" {{DOMException}}.
-		3. If |rule| is an <a spec=css-cascade-4>@import</a> rule and |sheet|'s [=constructed flag=] is set, throw "{{NotAllowedError}}" {{DOMException}}.
-		4. (The rest of the algorithm remains as in CSSOM)
+		3. (The rest of the algorithm remains as in CSSOM)
 	</dd>
 
-    <dt><dfn method for=CSSStyleSheet lt="replace(text)">replace(text)</dfn></dt>
+    <dt><dfn method for=CSSStyleSheet>replace(|text|)</dfn></dt>
     <dd>
 		1. Let |sheet| be the stylesheet on which this function is called on.
 		2. If |sheet|'s [=constructed flag=] is not set, or |sheet|'s [=disallow modification flag=] is set, throw a "{{NotAllowedError}}" {{DOMException}}.
 		3. Set |sheet|'s [=CSS rules=] to an empty list, and set |sheet|'s [=disallow modification flag=].
-		4. <a spec=css-syntax-3>Parse a list of rules</a> from {{text}}.
+		4. <a>Parse a list of rules</a> from |text|.
 			If it returned a list of rules,
 			assign the list as |sheet|'s [=CSS rules=];
-			otherwise,
-			set |sheet|'s [=CSS rules=] to an empty list.
-		5. If |sheet| contains one or more <a spec=css-cascade-4>@import</a> rules and |sheet|'s [=list of adopter documents=] contains any other {{Document}} than |sheet|'s [=constructor document=], throw a "{{NotAllowedError}}" {{DOMException}}.
-		6. Let |promise| be a promise.
-		7. In parallel, wait for loading of <a spec=css-cascade-4>@import</a> rules in |sheet| and any nested <a spec=css-cascade-4>@import</a>s from those rules.
-			* If any of them failed to load or resulted in a resource with a <a spec=html>Content-Type metadata</a> of anything other than <code>text/css</code>, reject |promise| with reason set to "{{NotAllowedError}}" {{DOMException}}.
-		    * Otherwise, resolve |promise| with |sheet| once all of them have finished loading. and unset |sheet|'s [=disallow modification flag=].
+		5. Let |promise| be a promise.
+		6. In parallel, wait for loading of <a>@import</a> rules in |sheet| and any nested <a>@import</a>s from those rules.
```

Comment:
Cross-reference in parallel for clarity?

Reply:
Also, this reads a bit as if waiting for @import only happens up to two levels of nesting.